plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/SsoAccessTokenProvider.kt

```diff
@@ -373,16 +376,43 @@ class SsoAccessTokenProvider(
             throw InvalidRequestException.builder().message(message).build()
         }
 
-        val registration = when (currentToken) {
-            is DeviceAuthorizationGrantToken -> loadDagClientRegistration()
-            is PKCEAuthorizationGrantToken -> loadPkceClientRegistration()
+        var registration: ClientRegistration? = null
+        try {
+            registration = when (currentToken) {
+                is DeviceAuthorizationGrantToken -> loadDagClientRegistration()
+                is PKCEAuthorizationGrantToken -> loadPkceClientRegistration()
+            }
+        } catch (e: Exception) {
+            val message = "Error loading client registration: ${e.message}"
+            sendRefreshCredentialsMetric(
+                currentToken,
+                reason = "Failed to load client registration",
+                reasonDesc = "Step: Load Registration - $message",
+                result = Result.Failed
+            )
+            throw InvalidClientException.builder().message(message).cause(e).build()
+        }
+
+        var isExpired = currentToken.expiresAt.isBefore(Instant.now(clock))
```

Comment:
Is this step required, can we key off the null registration that is sent over by the loading function?